src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,135 @@
+package org.apache.mesos.state;
+
+import com.google.protobuf.TextFormat;
+import org.apache.mesos.Protos.*;
+import org.apache.mesos.curator.CuratorStateStore;
+import org.apache.mesos.offer.TaskException;
+import org.apache.mesos.offer.TaskUtils;
+import org.apache.mesos.reconciliation.TaskStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState implements TaskStatusProvider {
```

Comment:
It looks like a lot of functions in this class are just performing datatype conversions against the underlying StateStore. Are those really needed or could they be omitted here?